service/elements_client.go

```diff
@@ -18,6 +18,14 @@ var (
 	elementsSyncAccess sync.Mutex
 )
 
+func isValidAmount(amount string) (valid bool) {
+	f, err := strconv.ParseFloat(amount, 64)
+	if err == nil && f != 0.0 {
+		valid = true
+	}
+	return
+}
+
```

Comment:
This function looks a bit off how about:

```suggestion
func isValidAmount(amount string) bool {
    amount = strings.TrimSpace(amount)
    f, err := strconv.ParseFloat(amount, 64)

    if err == nil && f > 0 {
        return true
    }
    return false
}
```